src/muler/utilities.py

```diff
@@ -240,3 +241,92 @@ def is_list(check_this):
     else:
         return False
 
+def estimate_slit_throughput_ABBA(y, x=None, slit_length=14.8, slit_width=1.0, PA=90.0, guiding_error=1.5, print_info=True):
+    """
+    Given a collapsed spatial profile long slit for a point (stellar) source nodded
+    ABBA along the slit, returns a numerical estimate of the fraction of light through the slit.
+    The A and B nods are fit with Moffat functions which are then projected from 1D to 2D and then
+    a mask is applied representing the slit and the the fraction of light in the PSFs inside the mask
+    are integrated to estimate the fraction of light that passes through the slit.
+
+    Parameters 
+    ----------
+    y: numpy array of floats
+        Array representing the spatial profile of the source on the slit.  It should be the PSF for
+        a point source nodded ABBA on the slit.
+    x: numpy array of floats (optional)
+        Array representing the spatial position along the slit in pixel space corrisponding to y.
+    slit_length: float
+        Length of the slit on the sky in arcsec.
+    slit_width: float
+        Width of the slit on the sky in arcsec.
+    PA: float
+        Position angle of the slit on the sky in degrees.  Measured counterclockwise from North to East.
+    guilding_error: float
+        Estimate of the guiding error in arcsec.  This smears out the PSF fits in the East-West direction.
+        This should be used carefully and only for telescopes on equitorial mounts.
+    print_info: bool
+        Print information about the fit.
+
+    Returns: 
+    ----------
+    Float: The fraction of light from the source estimated to pass through the slit.
+    """
+
+
+    slit_width_to_length_ratio = slit_width / slit_length
+    if x is None: #Generate equally spaced x array if it is not provided
+        ny = len(y)
+        x = (np.arange(ny) / ny) * slit_length
+    #Find maximum and minimum
+    i_max = np.where(y == np.nanmax(y))[0][0]
+    i_min = np.where(y == np.nanmin(y))[0][0]
+    if np.size(i_max) > 1: #Error catch for the rare event when two or more pixels match the max or min y values
+        i_max = i_max[0]
+    if np.size(i_min) > 1:
+        i_min = i_min[0]
+    #Fit 2 Moffat distributions to the psfs from A and B positions (see https://docs.astropy.org/en/stable/modeling/compound-models.html)
+    g1 = models.Moffat1D(amplitude=y[i_max], x_0=x[i_max], alpha=1.0, gamma=1.0)
+    g2 = models.Moffat1D(amplitude=y[i_min], x_0=x[i_min], alpha=1.0, gamma=1.0)
+    gg_init = g1 + g2
+    fitter = fitting.TRFLSQFitter()
+    gg_fit = fitter(gg_init, x, y)
+    if print_info:
+        print('FWHM A beam:', gg_fit[0].fwhm)
+        print('FWHM B beam:', gg_fit[1].fwhm)
+    #Numerically estimate light through slit

```

Comment:
Yes the assumption is that the Moffat function is always centered (in the dispersion direction) on the slit.  Obviously this is not a perfect assumption but it assumes on average the guiding software tends to keep things symmetrically centered over the slit.  The biggest issue is probably when using an off-slit guide star, so it is definitely a problem that can come up.

We could precompute "truncated Moffat2Ds", I don't see why not.  At this point though it's really going to have to be on the user to figure out where the true center, since writing some automated way to read in a series of SVC images and combine that with the PSFs seen on the spectrometer's detectors seems like an involved task.  Ideally this would be how it could be done though.

Reply:
I actually have tried making a "truncated" Moffat function in 1D by fixing the y axis and making it a parameter in the fit (stop by my office if you want to see the math).  Surprisingly it does not seem to do any better fitting the slit profile than an ordinary Moffat function.  Fitting two moffat functions, or a Gaussian and a Moffat, per beam seems to give me a much better fit, but then it's not obvious to me how to extract a FWHM from that.